cli/waiter/subcommands/show.py

```diff
@@ -0,0 +1,108 @@
+import json
+
+from tabulate import tabulate
+
+from waiter.format import format_mem_field, format_timestamp_string, format_field_name
+from waiter.querying import query_across_clusters, get_token
+from waiter.util import guard_no_cluster
+
+
+def tabulate_token(cluster_name, token, token_name):
+    """Given a token, returns a string containing tables for the fields"""
+    table = [['Cluster', cluster_name],
+             ['Owner', token['owner']]]
+    if token.get('name'):
+        table.append(['Name', token['name']])
+    if token.get('cpus'):
+        table.append(['CPUs', token['cpus']])
+    if token.get('mem'):
+        table.append(['Memory', format_mem_field(token)])
+    if token.get('ports'):
+        table.append(['Ports Requested', token['ports']])
+
+    explicit_keys = ('cmd', 'cpus', 'last-update-time', 'last-update-user', 'mem', 'name', 'owner', 'ports')
+    ignored_keys = ('cluster', 'previous', 'root')
+    for key, value in token.items():
+        if key not in (explicit_keys + ignored_keys):
+            table.append([format_field_name(key), value])
+
+    command = token.get('cmd')
+    if command:
+        token_command = f'Command:\n{command}'
+    else:
+        token_command = '<No command specified>'
+
+    if token.get('env') and len(token['env']) > 0:
+        environment = '\n\nEnvironment:\n%s' % '\n'.join(['%s=%s' % (k, v) for k, v in token['env'].items()])
+    else:
+        environment = ''
+
+    table_text = tabulate(table, tablefmt='plain')
+    last_update_time = format_timestamp_string(token['last-update-time'])
+    last_update_user = token['last-update-user']
+    return f'\n' \
+           f'=== Token: {token_name} ===\n' \
+           f'\n' \
+           f'Last Updated: {last_update_time} ({last_update_user})\n' \
+           f'\n' \
+           f'{table_text}\n' \
+           f'\n' \
+           f'{token_command}' \
+           f'{environment}'
+
+
+def show_data(cluster_name, data, format_fn, token_name):
+    """Iterates over the data collection and formats and prints each element"""
+    count = len(data)
+    if count > 0:
+        output = format_fn(cluster_name, data, token_name)
+        print(output)
+        print()
+    return count
+
+
+def get_token_on_cluster(cluster, token_name):
+    """Gets the token with the given name on the given cluster"""
+    token = get_token(cluster, token_name, include='metadata')
```

Comment:
Is this the token (i.e. name of the token) or the token data? Would it be clearer to name it `token_fields`?

Reply:
Renamed to `token_data`